sanic/mixins/routes.py

```diff
@@ -957,3 +990,18 @@ def visit_Return(self, node: Return) -> Any:
         HttpResponseVisitor().visit(node)
 
         return types
+
+    def _build_route_context(self, raw):
+        ctx_kwargs = {
+            key.replace("ctx_", ""): value
+            for key, value in raw.items()
+            if key.startswith("ctx_")
+        }
+        restricted = [
+            key for key in ctx_kwargs.keys() if key in RESTRICTED_ROUTE_CONTEXT
+        ]
+        if restricted:
+            raise AttributeError(
+                f"Cannot use restricted route context: {restricted}."
+            )
+        return HashableDict(ctx_kwargs)
```

Comment:
Should raise an exception (NameError?) if any kwargs without that prefix are found.

Reply:
Agreed. I was planning on doing something like that so we didn't accept just anything.